pkg/derivation/encode.go

```diff
@@ -17,30 +19,26 @@ var stringEscaper = strings.NewReplacer(
 	"\"", "\\\"",
 )
 
+func unsafeGetBytes(s string) []byte {
+	return (*[0x7fff0000]byte)(unsafe.Pointer(
```

Comment:
what does the `0x7fff0000` magic number do? Does this also work properly on 32 bit architectures, or these with different endianess?

Reply:
It's the max length of the byte slice, the version you commented on would have been broken with 32 bit, but the current version isn't.
Not sure about endianess here.

Ideally we'd bump the version in go.mod to 1.17 and use `unsafe.Slice`.

Reply:
Do it :-) We're only testing Go 1.16 and 1.17 right now - might want to use 1.17 and 1.18 there, and bump the requirements to 1.17.

Reply:
#75